src/components/Button/Button.component.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import "./Button.style.css";
+import Polygon from "../ImageComponents/FatArrow.icon";
+
+const Button = ({ children, styleForButton, icon }) => {
+  let buttonStyle = "";
+  let orientation = "";
+
+  switch (styleForButton) {
+    case "primary":
+      buttonStyle = "white";
+      break;
+
+    case "secondary":
+      buttonStyle = "red";
+
+      break;
+
+    case "seeMore":
+      buttonStyle = "seeMore";
+      orientation = "right";
+
+      break;
+
+    case "loadMore":
+      buttonStyle = "loadMore";
+      orientation = "bottom";
+
+      break;
+
+    case "sortBy":
+      buttonStyle = "sortBy";
+      orientation = "bottom";
+
+      break;
+
+    default:
+      buttonStyle = "regular";
+      break;
+  }
+
+  const classes = `button button--${buttonStyle}`;
+
+  return (
+    <button className={classes}>
+      {children}
+      {icon ? <Polygon orientation={orientation} url={icon} /> : null}
```

Comment:
You shouldn't have a specialized icon in your button. Pass it as a child instead. There is a special use case though, if you want to show a 'loading' state inside the button, then the Spinner icon can be placed inside the Button itself and shown using a prop 'inProgress'